### ISSUE TITLE
Two Datasets opened in parallel via OPeNDAP are loaded into same resource

### ISSUE THREAD

**Issue by HelenClifton** [CCI-Tools/cate#306]
### Expected behavior

If more than one dataset is loaded into memory via OPeNDAP at the same time, expect a separate resource to be created for each one.

### Actual behavior

The tasks panel indicates that cate is only creating one resource and both datasets are being loaded into it.

### Steps to reproduce the problem

1.	Clear the workspace or create a new one

2.	From ESA CCI Open Data Portal list, select esacci.AEROSOL.5-days.L3C.AEX.GOMOS.Envisat.aergom.2-19.r1 then click on “Download and/or open remote dataset”

3.	In the “Download Data Source” window:
•	Deselect time, region, variable constraints.  
•	Deselect “Download and make local data source”. 
•	Select “open dataset from remote”
•	Click “Open Remote”

![capture - parallel datasets opendap 1](https://user-images.githubusercontent.com/29569434/28823307-cb9e492e-76b4-11e7-853b-bbec431eee04.PNG)

4.	In Tasks panel, observe there is a new task displayed “Opening dataset res_1 from esacci.AEROSOL.5-days.L3C.AEX.GOMOS.Envisat.aergom.2-19.r1”

5.	From ESA CCI Open Data Portal list, select esacci.OZONE.mon.L3.NP.multi-sensor.multi-platform.MERGED.fv0002.r1 then click on “Download and/or open remote dataset”

6.	In the “Download Data Source” window:
•	Deselect time, region, variable constraints.  
•	Deselect “Download and make local data source”. 
•	Select “open dataset from remote”
•	Click “Open Remote”

![capture - parallel datasets opendap 2](https://user-images.githubusercontent.com/29569434/28823320-d55142aa-76b4-11e7-996a-04d4c5a46482.PNG)

7.	In Tasks panel, observe there is a new task displayed “Opening dataset res_1 from esacci.OZONE.mon.L3.NP.multi-sensor.multi-platform.MERGED.fv0002.r1”

![capture - parallel datasets opendap 3](https://user-images.githubusercontent.com/29569434/28823332-e383205a-76b4-11e7-8e21-94d75d82538c.PNG)

### Specifications 
Cate 0.9.0-dev.4
Windows 7 Professional

**Comment by HelenClifton** [comment]
Same happens if trying to open more than one local dataset at the same time

**Comment by forman** [comment]
User should provide reasonable, unique resource name

**Comment by forman** [comment]
@mzuehlke remember, 0.9.0-dev.5 in you've implemented a feature *Prevent a 2nd (and 3rd) invocation of the variable statistics computation*. This is another application of your fix. I'll try to generalize it for any operation invocation.

**Comment by forman** [comment]
For 1.0, it is sufficient to ensure unique resource names. Later we'll add a feature to let users provide their own resource names when opening datasets and applying operations.

**Comment by HelenClifton** [comment]
In cate-0.9.0-dev7 the functionality in the Download Data Source dialog to load into memory via OPeNDAP has been disabled.
In cate-0.9.0-dev7 it is still possible to open two local datasets in parallel and they will be assigned the same resource name. Will retest in 1.0.0.

**Comment by forman** [comment]
> In cate-0.9.0-dev7 the functionality in the Download Data Source dialog to load into memory via OPeNDAP has been disabled.
Oops, OPeNDAP should be used automatically if any constraints are used. @kbernat, no?

> In cate-0.9.0-dev7 it is still possible to open two local datasets in parallel and they will be assigned the same resource name. Will retest in 1.0.0.
Yes, yes, I have an idea...

**Comment by forman** [comment]
@HelenClifton I cannot reproduce your problem. What exactly have you done in order to open two datasets that have same resource names?

**Comment by mzuehlke** [comment]
@forman can it be that the fix has only been merged after the release of 0.9.0.dev7 ?

**Comment by forman** [comment]
Yes. Shame on me.

**Comment by forman** [comment]
Furthermore, the OPeNDAP option has accidentally been removed while resolving #366! I will reintroduce it.

**Comment by forman** [comment]
It's not true. OPeNDAP should be used automatically when you uncheck option "Download and make local data source". I will clarify this with a tooltip in 1.0.

Sorry for this confusion, too many activities in parallel.

**Comment by forman** [comment]
@HelenClifton it'd be great if you could take a look again in 1.0.0.dev1 on Monday

**Comment by HelenClifton** [comment]
@forman Confirmed FIXED in 1.0.0.dev2
